src/core/drive/page_snapshot.ts

```diff
@@ -24,7 +24,22 @@ export class PageSnapshot extends Snapshot<HTMLBodyElement> {
   }
 
   clone() {
-    return new PageSnapshot(this.element.cloneNode(true), this.headSnapshot)
+    const clonedElement = this.element.cloneNode(true)
+
+    const selectElements = this.element.querySelectorAll("select")
+    const clonedSelectElements = clonedElement.querySelectorAll("select")
+
+    for (const [index, source] of selectElements.entries()) {
+      for (const [optionIndex, option] of Array.from(source.options).entries()) {
+        clonedSelectElements[index].options[optionIndex].selected = option.selected
+      }
+    }
```

Comment:
FYI, the following approach is around 3x to 4x faster.  Not an issue for small selects, but if the page contains large selects (e.g. Country Selector) it adds up.

```javascript
for (const [index, source] of selectElements.entries()) {
  [...clonedSelectElements[index].selectedOptions].forEach(option => { option.selected = false });
  [...source.selectedOptions].forEach(option => clonedSelectElements[index].options[option.index].selected = true)
}
```

Reply:
Would it be even faster if the `[...]` syntax were replaced with a `for ... of` loop? I can open a follow-up PR.

Reply:
I've opened https://github.com/hotwired/turbo/pull/669. Thank you!